homeassistant/components/esphome/bluetooth/client.py

```diff
@@ -540,3 +540,15 @@ async def stop_notify(
         # to be consistent with the behavior of the BlueZ backend
         if coro := self._notify_cancels.pop(characteristic.handle, None):
             await coro()
+
+    def __del__(self) -> None:
+        """Destructor to make sure the connection state is unsubscribed."""
+        if self._cancel_connection_state:
+            _LOGGER.warning(
+                "%s: %s - %s: ESPHomeClient bleak client was not properly disconnected before destruction",
+                self._source,
+                self._ble_device.name,
+                self._ble_device.address,
+            )
+        if not self._hass.loop.is_closed():
+            self._hass.loop.call_soon_threadsafe(self._unsubscribe_connection_state)
```

Comment:
Can you still call things on something that is going to be deleted ? Or will this cancel deletion ?

Reply:
The call here is very likely to fail since it's likely the object will be none at this point depending on the order of the garage collection but we are past the point of no return here so it's only a best effort to try to cleanup the leak.

The logger should work regardless which should hopefully make use aware of future issues so we can close them properly as we really should never get here if we don't make a mistake in the implementation